src/PresentationalComponents/Email/Email.js

```diff
@@ -1,53 +1,80 @@
 import React, { useState, useEffect } from 'react';
 import './email.scss';
+import { useSelector, useDispatch } from 'react-redux';
 import { formFieldsMapper, layoutMapper } from '@data-driven-forms/pf4-component-mapper';
 import { Main, PageHeader, PageHeaderTitle, Skeleton } from '@redhat-cloud-services/frontend-components';
 import { Button, Card, CardBody, Stack, StackItem, Flex, FlexItem, FlexModifiers, CardHeader } from '@patternfly/react-core';
 import FormRender from '@data-driven-forms/react-form-renderer';
 import PropTypes from 'prop-types';
-import { DESCRIPTIVE_CHECKBOX, DescriptiveCheckbox, LOADER, Loader } from '../../SmartComponents/FormComponents';
+import { DESCRIPTIVE_CHECKBOX, DATA_LIST, LOADER, DescriptiveCheckbox, DataListLayout, Loader } from '../../SmartComponents/FormComponents';
+import config from '../../config.json';
+import { emailPreferences, register } from '../../store';
+import { getEmailSchema, saveEmailValues } from '../../actions';
 
-const FormButtons = ({ submitting, valid, pristine, onCancel }) => (
+const FormButtons = ({ submitting, pristine, onCancel }) => (
     <div>
         <Button
             type="submit"
-            isDisabled={ submitting || !valid }
+            isDisabled={ submitting || pristine }
             style={ { marginRight: 16 } }
             variant="primary">Save</Button>
         <Button
             variant="link"
             isDisabled={ pristine }
-            onClick={ onCancel }>Cancel</Button>
+            onClick={ onCancel }>
+                Cancel
+        </Button>
     </div>
 );
 
+const getSchema = (app) => {
+    return !app || !app.loaded ? [{
+        fields: [{
+            component: LOADER
+        }]
+    }] : app?.schema;
```

Comment:
I think that the `app?.schema` is not necessary here. You check the existence of the `app` variable on the top of this function.